code/game/objects/effects/decals/Cleanable/fuel.dm

```diff
@@ -34,6 +34,18 @@
 		qdel(other)
 	fuel_spread()
 
+/obj/effect/decal/cleanable/liquid_fuel/Crossed(atom/movable/AM)
+	. = ..()
+	if(isitem(AM))
+		var/obj/item/igniter = AM
+		if(igniter.damtype != BURN)
+			return
+		addtimer(CALLBACK(src, .proc/crossed_ignite, igniter), 1) //Just to make sure it wasn't thrown over
```

Comment:
Absolutely not, this is a hot proc and spamming timers on it only makes it worse

Reply:
I figured it would be alright because a: there's few to zero fuel puddles most of the time, b: most things that cross it won't spawn a timer, and c: most things that do cross it and spawn a timer also delete the puddle.